shell/common/rasterizer.h

```diff
@@ -20,78 +20,367 @@
+  ///             protocol for instrumentation tools running on the host).
+  ///
+  /// @param[in]  type           The type of the screenshot to gather.
+  /// @param[in]  base64_encode  Whether Base 64 encoding must be applied to the
+  ///                            data after a screenshot has been captured.
+  ///
+  /// @return     A non-empty screenshot if one could be captured. A screenshot
+  ///             capture may fail if there were no layer trees previously
+  ///             rendered by this rasterizer, or, due to an unspecified
+  ///             internal error. Internal error will be logged to the console.
+  ///
   Screenshot ScreenshotLastLayerTree(ScreenshotType type, bool base64_encode);
 
-  // Sets a callback that will be executed after the next frame is submitted to
-  // the surface on the GPU task runner.
+  //----------------------------------------------------------------------------
+  /// @brief      Sets a callback that will be executed when the next layer tree
+  ///             in rendered to the on-screen surface. This is used by
+  ///             embedders to listen for one time operations like listening for
+  ///             when the first frame is rendered so that they may hide splash
+  ///             screens.
+  ///
+  ///             The callback is only executed once and dropped on the GPU
+  ///             thread when executed (lambda captures must be able to deal
+  ///             with the threading repercussions of this behavior).
+  ///
+  /// @param[in]  callback  The callback to execute when the next layer tree is
+  ///                       rendered on-screen.
+  ///
   void SetNextFrameCallback(fml::closure callback);
 
+  //----------------------------------------------------------------------------
+  /// @brief      Returns a pointer to the compositor context used by this
+  ///             rasterizer. This pointer will never be `nullptr`.
+  ///
+  /// @return     The compositor context used by this rasterizer.
+  ///
   flutter::CompositorContext* compositor_context() {
     return compositor_context_.get();
   }
 
+  //----------------------------------------------------------------------------
+  /// @brief      Skia has no notion of time. To work around the performance
+  ///             implications of this, it may cache GPU resources to reference
+  ///             them from one frame to the next. Using this call, embedders
+  ///             may set the maximum bytes cached by Skia in its caches
+  ///             dedicated to on-screen rendering.
+  ///
+  /// @attention  This cache setting will be invalidated when the surface is
+  ///             torn down via `Rasterizer::Teardown`. This call must be made
+  ///             again with new limits after surface re-acquisition.
+  ///
+  /// @attention  This cache does not describe the entirety of GPU resources
+  ///             that may be cached. The raster cache also holds very large GPU
```

Comment:
Maybe add a reference to `|RasterCache|` (so they know that raster cache is not just a concept, but an actual class)?

Reply:
Done.